drivers/ad5770r/ad5770r.c

```diff
@@ -0,0 +1,650 @@
+ * and autoincrements for each aditional byte in the transfer.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @param count - Number of bytes to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_read_multiple(struct ad5770r_dev *dev,
+				      uint8_t reg_addr,
+				      uint8_t *reg_data,
+				      uint16_t count)
+{
+	uint8_t buf[1024];
+	uint16_t i;
+	int32_t ret;
+
+	/* Check if multibyte enabled */
+
+	buf[0] = AD5770R_REG_READ(reg_addr);
+	memset(&buf[1], 0x00, count);
+
+	ret = spi_write_and_read(dev->spi_desc, buf, count + 1);
+
+	for (i = 0; i < count; i++)
+		reg_data[i] = buf[i+1];
+
+	return ret;
+}
+/**
+ * Write to device.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_write(struct ad5770r_dev *dev,
+			      uint8_t reg_addr,
+			      uint8_t reg_data)
+{
+	int32_t ret;
+	uint8_t buf[2];
+
+	buf[0] = AD5770R_REG_WRITE(reg_addr);
+	buf[1] = reg_data & 0xFF;
+
+	ret = spi_write_and_read(dev->spi_desc, buf, 2);
+
+	return ret;
+}
+
+/**
+ * Multibyte write from device. A register write begins with the address
+ * and autoincrements for each additional byte in the transfer.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - The register data.
+ * @param count - Number of bytes to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_reg_write_multiple(struct ad5770r_dev *dev,
+				       uint8_t reg_addr,
+				       uint8_t *reg_data,
+				       uint16_t count)
+{
+	int32_t ret;
+	uint8_t data[1024];
+
+	data[0] = AD5770R_REG_WRITE(reg_addr);
+	memcpy(&data[1], reg_data, count);
+
+	ret = spi_write_and_read(dev->spi_desc, data, count + 1);
+
+	return ret;
+}
+
+/**
+ * SPI write to device using a mask.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - The mask.
+ * @param data - The register data.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_spi_write_mask(struct ad5770r_dev *dev,
+			       uint8_t reg_addr,
+			       uint32_t mask,
+			       uint8_t data)
+{
+	uint8_t reg_data;
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_read(dev, reg_addr, &reg_data);
+	reg_data &= ~mask;
+	reg_data |= data;
+	ret |= ad5770r_spi_reg_write(dev, reg_addr, reg_data);
+
+	return ret;
+}
+
+/**
+ * Set device spi settings.
+ * @param dev - The device structure.
+ * @param spi_settings - The structure that contains the device spi
+ *		       parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_device_spi(struct ad5770r_dev *dev,
+			       struct device_spi_settings spi_settings)
```

Comment:
pass this via const pointer;
```
int32_t ad5770r_set_device_spi(struct ad5770r_dev *dev,
			       const struct device_spi_settings *spi_settings)
```

the way this is done now, copies the entire `struct device_spi_settings` on the call-stack